source/controllers/src/ControllerFactory.cpp

```diff
@@ -88,7 +88,8 @@ std::shared_ptr<IController<JointState>> ControllerFactory<JointState>::create_c
     CONTROLLER_TYPE type, const std::list<std::shared_ptr<ParameterInterface>>& parameters,
     const robot_model::Model& robot_model
 ) {
-  auto ctrl = ControllerFactory<JointState>::create_controller(type, parameters, robot_model.get_number_of_joints());
+  auto ctrl =
```

Comment:
Sorry I keep being confused about this, the configuration dimension is the number of variables necessary to describe the joint state, right? I'm not sure that our idea of controllers is compatible with anything else than revolute and prismatic joint or how would an impedance controller look like for a continuous joint? It doesn't work because the two numbers are coupled no?

Reply:
yes, it would fail. I only supported forward kinematics and velocities, so everything else is still incompatible (theoretically and in practice). 

So perhaps we should completely fail exactly in this place, but indeed here it should also be number of joints, not configuration parameters